### PR TITLE
 Please enter the commit message for your changes. Lines starting

### PR DESCRIPTION
**Pull request by jitendra-khasdev**
Adding support for cd <No Args>.

### CODE REVIEW

File: input_parser.c [hunk 1]
```diff
@@ -85,6 +105,20 @@ main()
 */ 
 void change_dir(char *argv[]) 
 {
```
Comment (pankaj1jul) [inline input_parser.c]: Add your name & Email Address as a developer of this function.

File: input_parser.c [hunk 2]
```diff
@@ -85,6 +105,20 @@ main()
 */ 
 void change_dir(char *argv[]) 
 { 
+<<<<<<< HEAD
+
+int fp;
+int c;
```
Comment (pankaj1jul) [inline input_parser.c]: Put "c and i " into one declaration line. Add also repeat declaration here.

File: input_parser.c [hunk 3]
```diff
@@ -85,6 +105,20 @@ main()
...
+int i=0;
+static char buffer[200];
+int End_File_Position;
+int Current_Position=1;
+int flag=0;
```
Comment (pankaj1jul) [inline input_parser.c]: Flag and repeat also will go with on line number 111, with c and i.

File: input_parser.c [hunk 4]
```diff
@@ -85,6 +105,20 @@ main()
 */ 
 void change_dir(char *argv[]) 
 { 
+<<<<<<< HEAD
+
+int fp;
+int c;
+int i=0;
+static char buffer[200];
+int End_File_Position;
```
Comment (pankaj1jul) [inline input_parser.c]: Keep these 2 position flags in one line.

File: input_parser.c [hunk 5]
```diff
@@ -101,4 +135,118 @@ void change_dir(char *argv[])
...
+                       }					
+                
+ 	      } 
+         }
+       
+}     
+
+
+
+int search_path(char *ptr,int size,int userid)
```
Comment (pankaj1jul) [inline input_parser.c]: If you copy buffer size 200 bytes. The conditions you have to handle like:
1) if 199th character falls on 2nd character of the UID.
2) If  199th character falls on the delimiter after the UID, and it meets to hit or miss, Do you save the state in permanent variable.

Solutions:
Keeping a static array and static index for saving last line, which has been read from buffer.
Keeping your status variable static or global one.

**Comment by jitendra-khasdev** [comment]
This is my new cd support for home directory with fixed bugs which was in last version

File: input_parser.c [hunk 6]
```diff
@@ -101,4 +133,144 @@ void change_dir(char *argv[])
...
+                  
+ 	               while(i < End_File_Position){
+                    	
+				for(k=0;k<200;k++) buffer[k]='\0';    //initiallzation of buffer
+
+                    	 		if(count =  read( fp, buffer, sizeof(buffer) )   ){
+
+                       				for(k=0,j=0;k<200;k++,j++){
+
+                                           	    temp[j]=*(buffer +k);
```
Comment (pankaj1jul) [inline input_parser.c]: Good work!!!!!

File: input_parser.c [hunk 7]
```diff
@@ -101,4 +133,144 @@ void change_dir(char *argv[])
...
+                                                    
+						    if( *(buffer+k) == '\n' ){
+                                                    	       if( search_path(temp,getuid()) != 0){
```
Comment (pankaj1jul) [inline input_parser.c]: Why do you need 500 bytes on temp, when you are just reading 200 bytes?

File: input_parser.c [hunk 8]
```diff
@@ -101,4 +133,144 @@ void change_dir(char *argv[])
...
+char *temp_arg[1];
+
+                      	
+	 for(k=0;k<100;k++) temp[k]='\0';
+         
+	 while(*(ptr) != '\n')
+         {
+                          
+                         
+                	  // count = 0;   // intializing delimiter for every new line
```
Comment (pankaj1jul) [inline input_parser.c]: Explain this condition about count on white board, on Monday, 30/06/2014.

**Comment by jitendra-khasdev** [comment]
basic single stage pipe support and cd bugs fix

**Comment by jitendra-khasdev** [comment]
sir i have indented it properly

**Comment by jitendra-khasdev** [comment]
bug fix in cd support and provide ~ sign support

**Pull request merged**